pkg/process/checks/net.go

```diff
@@ -85,12 +90,102 @@ func (c *ConnectionsCheck) Run(cfg *config.AgentConfig, groupID int32) ([]model.
 	// Resolve the Raddr side of connections for local containers
 	LocalResolver.Resolve(conns)
 
+	// Get the container and process relationship from either the process or container checks
+	ctrsByPid := ctrIDsByPIDs(connectionPIDs(conns.Conns))
+
+	resolveLoopbackConnections(conns.Conns, ctrsByPid)
+
 	tel := c.diffTelemetry(conns.Telemetry)
 
 	log.Debugf("collected connections in %s", time.Since(start))
 	return batchConnections(cfg, groupID, c.enrichConnections(conns.Conns), conns.Dns, c.networkID, tel), nil
 }
 
+func resolveLoopbackConnections(conns []*model.Connection, ctrsByPid map[int32]string) {
+
+	ctrsByLaddr := make(map[model.ContainerAddr]string)
+	ctrsByLoopback := make(map[addrWithNS]string)
```

Comment:
It might be clearner to combine this into a single (pre-sized) map and use 0 as the netns for n on-loopback addrs  ?

Reply:
Yes, done.